test/src/e2e_vm_tests/test_programs/supertraits/src/main.sw

```diff
@@ -0,0 +1,179 @@
+            r3: bool
+        }
+    }
+}
+
+impl Color for PrimaryColor {
+    // Method from Color
+    fn rgb(self) -> Rgb {
+        if self == PrimaryColor::Red {
+            Rgb {
+                red: 255,
+                blue: 0,
+                green: 0,
+            }
+        } else if self == PrimaryColor::Green {
+            Rgb {
+                red: 0,
+                blue: 0,
+                green: 255,
+            }
+        } else if self == PrimaryColor::Blue {
+            Rgb {
+                red: 0,
+                blue: 255,
+                green: 0,
+            }
+        }
+        else {
+            Rgb {
+                red: 0,
+                green: 0,
+                blue: 0,
+            }
+        }
+    }   
+
+    // Method from ColorBase1
+    fn rgb1(self) -> Rgb {
+        if self == PrimaryColor::Red {
+            Rgb {
+                red: 255,
+                blue: 0,
+                green: 0,
+            }
+        } else if self == PrimaryColor::Green {
+            Rgb {
+                red: 0,
+                blue: 0,
+                green: 255,
+            }
+        } else if self == PrimaryColor::Blue {
+            Rgb {
+                red: 0,
+                blue: 255,
+                green: 0,
+            }
+        }
+        else {
+            Rgb {
+                red: 0,
+                green: 0,
+                blue: 0,
+            }
+        }
+    }
+
+    // Method from ColorBase2
+    fn rgb2(self) -> Rgb {
+        if self == PrimaryColor::Red {
+            Rgb {
+                red: 255,
+                blue: 0,
+                green: 0,
+            }
+        } else if self == PrimaryColor::Green {
+            Rgb {
+                red: 0,
+                blue: 0,
+                green: 255,
+            }
+        } else if self == PrimaryColor::Blue {
+            Rgb {
+                red: 0,
+                blue: 255,
+                green: 0,
+            }
+        }
+        else {
+            Rgb {
+                red: 0,
+                green: 0,
+                blue: 0,
+            }
+        }
+    }
+}
+
+fn main() -> u32 {
+    // Test all methods from Color1
+    let first_color_1: PrimaryColor = PrimaryColor::Green;
+    let first_rgb_1: Rgb = first_color_1.rgb();
+    let first_color_2: PrimaryColor = PrimaryColor::Green;
+    let first_rgb_2: Rgb = first_color_2.rgb_wrapper();
+
+    // Test all methods from Color2
+    let second_color_1: PrimaryColor = PrimaryColor::Blue;
+    let second_rgb_1: Rgb = second_color_1.rgb1();
+    let second_color_2: PrimaryColor = PrimaryColor::Blue;
+    let second_rgb_2: Rgb = second_color_2.rgb1_wrapper();
+
+    // Test all methods from Color
+    let third_color_1: PrimaryColor = PrimaryColor::Red;
+    let third_rgb_1: Rgb = third_color_1.rgb2();
+    let third_color_2: PrimaryColor = PrimaryColor::Red;
+    let third_rgb_2: Rgb = third_color_2.rgb2_wrapper();
+
```

Comment:
These tests are definitely checking that everything compiles fine, but ideally they'd also confirm that the right calls are being made and the correct results are returned.

Reply:
My two new tests should do a better job at checking that the right calls are being made.